### PR TITLE
Sharmila

### PR DESCRIPTION
**Pull request by sharmiamala**

### CODE REVIEW

File: qcore/test/runscript.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+DIRS='test_*'
+curdir=$PWD
+for DIR in $DIRS
+do
+cd ${curdir}/${DIR}
+pytest --junitxml ${DIR}.xml
+cp -r ./${DIR}.xml /home/aas105/.jenkins/workspace/qcore-tests/${DIR}.xml
```
Comment (sungeunbae) [inline qcore/test/runscript.sh]: Hard-coded path?

File: qcore/test/runscript.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+done
+LOG=/home/aas105/.jenkins/jobs/qcore-tests/builds/$BUILD_NUMBER/log
```
Comment (sungeunbae) [inline qcore/test/runscript.sh]: /home/aas105?

File: qcore/test/runscript.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+sed -i 's/$/<br>/' ${LOG}
```
Comment (sungeunbae) [inline qcore/test/runscript.sh]: Leave a comment and describe what this line is supposed to do

File: qcore/test/test_gen_coords/test_gen_coords.py [hunk 4]
```diff
@@ -18,17 +18,22 @@
...
+
+SYMLINK_PATH = os.path.join(os.path.abspath(os.path.dirname(__file__)),INPUT_FILENAME)
 DIR_NAME = (os.path.join("/home/",getpass.getuser(),("tmp_" + os.path.basename(__file__)[:-3] + '_' + ''.join(str(datetime.now()).split())).replace('.', '_')).replace(
             ':', '_'))
 # print "PATH_TO_NEW_OUTDIR: ", DIR_NAME
-PATH_FOR_PRG_TOBE_TESTED = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(os.getcwd())), "gen_coords.py"))
-# print "PATH_FOR_PRG_TOBE_TESTED: ",PATH_FOR_PRG_TOBE_TESTED
+# PATH_FOR_PRG_TOBE_TESTED = os.path.join(os.path.abspath(os.path.dirname(__file__)), "../gen_coords.py")
+# print "first PATH_FOR_PRG_TOBE_TESTED: **** ",PATH_FOR_PRG_TOBE_TESTED
+prg_path = "/home/"+getpass.getuser()+"/qcore/qcore/gen_coords.py"
```
Comment (sungeunbae) [inline qcore/test/test_gen_coords/test_gen_coords.py]: I'm not sure what this is doing.  gen_cords under /home/USER/qcore???

**Comment by sungeunbae** [comment]
In general, there are many issues with the path. When a pull request is accepted, your code will go into the main codebase. Can you expect hard-coded path, "aas105" or some assumptions that qcore will be under /home/USER/ to be valid?

**Comment by sharmiamala** [comment]
changed the hardcoded path. Now the test runs in tester user (which is a common user account for all of us)

File: qcore/test/test_gen_coords/test_gen_coords.py [hunk 5]
```diff
@@ -18,17 +18,21 @@
...
+
+SYMLINK_PATH = os.path.join(os.path.abspath(os.path.dirname(__file__)),INPUT_FILENAME)
 DIR_NAME = (os.path.join("/home/",getpass.getuser(),("tmp_" + os.path.basename(__file__)[:-3] + '_' + ''.join(str(datetime.now()).split())).replace('.', '_')).replace(
             ':', '_'))
 # print "PATH_TO_NEW_OUTDIR: ", DIR_NAME
-PATH_FOR_PRG_TOBE_TESTED = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(os.getcwd())), "gen_coords.py"))
-# print "PATH_FOR_PRG_TOBE_TESTED: ",PATH_FOR_PRG_TOBE_TESTED
+# PATH_FOR_PRG_TOBE_TESTED = os.path.join(os.path.abspath(os.path.dirname(__file__)), "../gen_coords.py")
+# print "first PATH_FOR_PRG_TOBE_TESTED: **** ",PATH_FOR_PRG_TOBE_TESTED
+PATH_FOR_PRG_TOBE_TESTED = os.path.abspath("/home/aas105/qcore/qcore/gen_coords.py")
```
Comment (sungeunbae) [inline qcore/test/test_gen_coords/test_gen_coords.py]: aas105?

File: qcore/test/test_xyts/test_xyts.py [hunk 6]
```diff
@@ -9,9 +9,9 @@
 import shutil
 import errno
 
-XYTS_ACTUAL_PATH ="/nesi/projects/nesi00213/sample_data/test_qcore/xyts.e3d"
+XYTS_ACTUAL_PATH =os.path.abspath("/nesi/projects/nesi00213/sample_data/test_qcore/xyts.e3d")
```
Comment (sungeunbae) [inline qcore/test/test_xyts/test_xyts.py]: This path only works for Hypocentre and is not part of qcore lib?

**Comment by sharmiamala** [comment]
Comments addressed. Also checked again with jenkins it works fine.

**Pull request merged**